standardhealth/fhir-mapper
# Initial Epic mapper

PR includes a new mapper for Epic that only deals with AllergyIntolerance for now (will be extended in the future). We were encountering a case where some of the test data from the environment had `AllergyIntolerance`s without a `substance.coding`. This mapper adds a coding to `AllergyIntolerance` as well as the relevant mCODE profiile.

To Test:

Use the CLI to map the example test fixture and inspect the results:

``` bash
yarn map OpenEpicMapper src/__test__/fixtures/open-epic-bundle.json ./result.json
```

result.json should have an `AllergyIntolerance` profile as well as a `substance.coding` (right now the code just defaults to the `substance.text`

Run the unit tests:

``` bash
yarn test
```

Test fixtures from here: https://open.epic.com/Clinical/FHIR?whereTo=allergy
